src/idlcxx/tests/generator/bounded_sequence_impl.cpp.in

```diff
@@ -88,3 +95,17 @@ size_t s::read_struct(const void *data, size_t position)
   return position;
 }
 
+size_t s::read_struct_swapped(const void *data, size_t position)
+{
+  position += (4 - (position&0x3))&0x3;  //alignment
+  uint32_t _se0 = *reinterpret_cast<const uint32_t*>(static_cast<const char*>(data)+position);  //number of entries in the sequence
+  org::eclipse::cyclonedds::topic::bswap_32(&_se0);
+  position += 4;  //moving position indicator
+  mem().assign(reinterpret_cast<const int32_t*>(static_cast<const char*>(data)+position),reinterpret_cast<const int32_t*>(static_cast<const char*>(data)+position)+_se0); //reading bytes for member: mem()
```

Comment:
I'm sure there is a cleaner way of generating this (perhaps something like a set of templated functions to deal with the various types you can encounter, then just invoke those for each of the fields?)

Reply:
I was thinking of the same thing, since these kinds of operations stook out to me like sore thumb.